### PR TITLE
Reorganize macro buttons

### PR DESCRIPTION
**Pull request by SoshJam**
## Description

Added functionality to rearrange the order in which a macro executes its assigned actions. Create a macro button with more than one action, then click and drag the actions to rearrange them.

## Related Issue

Closes https://github.com/Thorium-Sim/thorium/issues/3201

## Screenshots (if appropriate):

- [ ] I submitted a pull request or created an issue for documenting this
      feature on the [Thorium Docs](https://github.com/Thorium-Sim/thorium-docs)
      repo. (Include the issue or pull request url below.)

### CODE REVIEW

File: server/events/eventMacro.js [hunk 1]
```diff
@@ -113,3 +113,10 @@ App.on("triggerMacroButton", ({simulatorId, configId, buttonId}) => {
   if (macros.length > 0)
     App.handleEvent({simulatorId, macros}, "triggerMacros");
 });
+
+// Reorder Macro Action by Sam Hill
```
Comment (alexanderson1993) [inline server/events/eventMacro.js:117]: There's no need to leave your signature everywhere - it's already included as part of the git history.

File: src/containers/FlightDirector/MacroButtons/macroConfig.js [hunk 2]
```diff
@@ -104,6 +106,46 @@ const ActionList = ({
...
+            onSortEnd={({oldIndex, newIndex}) =>
+              action({
+                variables: {configId, id, oldIndex, newIndex},
+              })
+            }
+          />
+        )}
+      </Mutation>
+
+      {/* Old macro action list
```
Comment (alexanderson1993) [inline src/containers/FlightDirector/MacroButtons/macroConfig.js:148]: If you've commented out all of the old code that you've replaced, you should probably just delete that code anyway.

**Review by alexanderson1993 — CHANGES_REQUESTED** [review]
The code that you've written is good, but I have to reject this PR as it is right now for two reasons.

- Not a big deal, but you shouldn't be putting your name all over the code. Add yourself to the .all-contributorsrc file, include yourself in the credits in the src/components/client/credits.ts file, but don't put your name next to every piece of code you write.


- This is the biggest reason: Your PR makes it possible to reorganize the actions, not the buttons themselves. The linked issue requests the ability to reorder the buttons. In the attached screenshot, the list on the left is the one that should be reordered, not the one on the left.


Screen Shot 2022-02-11 at 7 55 34 PM

Take care of those two things and I'd be happy to merge this.

**Comment by SoshJam** [comment]
Changes should be complete.

**Comment by alexanderson1993** [comment]
LGTM. Nice work.

**Pull request merged**